cmd/geth/snapshot.go

```diff
@@ -43,7 +43,7 @@ var (
 	emptyRoot = common.HexToHash("56e81f171bcc55a6ff8345e692c0f86e5b48e01b996cadc001622fb5e363b421")
 
 	// emptyCode is the known hash of the empty EVM bytecode.
-	emptyCode = crypto.Keccak256(nil)
+	emptyCode = poseidon.EmptyCodeHash.Bytes()
```

Comment:
so i think this should be reverted later? Since we decide to keep Keccak256(nil) as emptyCode hash?

Reply:
ok i see with this special check in poseidon code hash, it works.

```
        if len(code) == 0 {
		return crypto.Keccak256Hash(nil)
	}
```